### ISSUE TITLE
Provision on AKS fails 

### ISSUE THREAD

**Issue by SbgMartin** [kyma-project/cli#656]
**Description**

Provisioning based on the guide https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-prepare-the-cluster somehow doesn't work. Despite handing over a proper TOML file or not I always receive this error message:

`Error: expected "service_principal.0.client_id" to not be an empty string, got

  on ../../../.kyma/clusters/azure/middleearth/isengard/aks.tf line 34, in resource "azurerm_kubernetes_cluster" "k8s":
  34: resource "azurerm_kubernetes_cluster" "k8s" {

Error: expected "service_principal.0.client_secret" to not be an empty string, got

  on ../../../.kyma/clusters/azure/middleearth/isengard/aks.tf line 34, in resource "azurerm_kubernetes_cluster" "k8s":
  34: resource "azurerm_kubernetes_cluster" "k8s" {`

I am using Kyma-cli for OS X and tried with stable version 1.15.1 first. Then I tried pre-release 1.17.0-rc3 with more or less the same result.

**Expected result**

Cluster is provisioned on AKS

**Actual result**

Error message as described above and process abort.

**Steps to reproduce**

Follow the commands given on the guide:

1. Create TOML file

CLIENT_ID = {YOUR_CLIENT_ID}
CLIENT_SECRET = {YOUR_CLIENT_SECRET}
SUBSCRIPTION_ID = {YOUR_SUBSCRIPTION_ID}
TENANT_ID = {YOUR_TENANT_ID}

2. Create AKS cluster

kyma provision aks -c {YOUR_CREDENTIALS_FILE_PATH} -n $CLUSTER_NAME -p $RS_GROUP -l $REGION

Even if the path points to a non-existent TOML file, I receive the same error message. Instead I would expect to obtain an input validation error (I guess I saw it in the coding). In the end the Terraform file is obviously not populated based on the missing file handling.

**Troubleshooting**

**Comment by a-thaler** [comment]
Hi @SbgMartin,
have you quoted the values in your TOML file?

We should have some more specific examples on it, as we recently did for the install command https://kyma-project.io/docs/master/cli/overview/#details-kyma-cli-command-usage-examples-install-kyma

**Comment by SbgMartin** [comment]
Hi @a-thaler ,

thanks for the hint. Actually I tried with toml file's content like this:

`CLIENT_ID = {100f016c-43f1-418z-afui-7e9e7d1c679g}`
`CLIENT_ID = 100f016c-43f1-418z-afui-7e9e7d1c679g`

But your hint was the one which finally made it work:

`CLIENT_ID = "100f016c-43f1-418z-afui-7e9e7d1c679g"`

Thanks for this.

By the way, even if no valid file is given to `kyma provision `command no error comes back. Maybe input validation could be enhanced that at least it's being checked wether a readable file has been passed.

Thanks again & kind regards
Martin

**Comment by a-thaler** [comment]
You are right, we should validate that a TAML file is passed. Furthermore the docu must be improved to have the quotes always. Will re-open to remember it.

**Comment by clebs** [comment]
Hi @SbgMartin,
we have changed the credentials file format for AKS provisioning to `JSON`, which will cause less confusion. We already had other inquiries regarding the TOML format.

Additionally, we have added validation following your feedback. Now wrong file paths or non parseable files will display the according error and provisioning will not be attempted.

Kyma documentation also has beed updated to reflect this change on latest master:
https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-a-cluster-prepare-the-cluster
